modules/core/shared/src/main/scala/gem/TargetEnvironment.scala

```diff
@@ -4,25 +4,66 @@
 package gem
 
 import cats.Eq
-import monocle.macros.Lenses
-
+import cats.implicits._
+import gem.enum.Instrument
 import scala.collection.immutable.TreeSet
 
-/** Collection of targets associated with an observation.
-  */
-@Lenses final case class TargetEnvironment(
-  asterism: Option[Asterism],
-  /* guide stars, */
-  userTargets: TreeSet[UserTarget]
-)
+/** Collection of targets associated with an observation. */
+sealed trait TargetEnvironment {
+
+  type I <: Instrument with Singleton
+
+  val asterism: Option[Asterism.Aux[I]]
+
+  /** Guide stars. */
```

Comment:
Actually, guide stars are still pending.  The user targets are for blind offsetting, etc.

When guide stars are added, they too will have an instrument dependency.

Reply:
This comment was left over from the original case class. I'll remove it.